src/main/java/org/folio/gobi/Mapper.java

```diff
@@ -144,632 +124,543 @@ public Mapper(Map<Mapping.Field, DataSourceResolver> mappings) {
     return future;
   }
 
-  private void mapMetaData(List<CompletableFuture<?>> futures, Metadata metaData, Document doc) {
-    if (mappings.containsKey(Mapping.Field.CREATED_DATE)) {
-      futures.add(mappings.get(Mapping.Field.CREATED_DATE)
-         .resolve(doc)
-         .thenAccept(o -> metaData.setCreatedDate((Date) o))
-         .exceptionally(Mapper::logException));
-   }
+  /**
+   * This method handles all the different fields that are dependent on Vendor record to be fetched from the vendors-storage
+   *
+   * @param futures
+   * @param eresource
+   * @param physical
+   * @param doc
+   */
+  private void mapVendorDependentFields(List<CompletableFuture<?>> futures, Eresource eresource, Physical physical,
+      CompositePurchaseOrder compPo, Claim claim, Document doc) {
+    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR))
+        .ifPresent(field -> futures.add(field.resolve(doc)
+              .thenAccept(o -> {
+                if (null != o) {
+                  Vendor vendor = (Vendor) o;
+                  compPo.setVendor(vendor.getId());
+                //Map Receipt dates based on the order format type
+                  LocalDateTime dt = LocalDateTime.from(new Date().toInstant().atOffset(ZoneOffset.UTC));
+                  Optional.ofNullable(mappings.get(Mapping.Field.PO_LINE_ORDER_FORMAT))
+                  .ifPresent(orderformat -> orderformat.resolve(doc)
+                        .thenAccept(format->{
+                            if(((String) format).equalsIgnoreCase(CompositePoLine.OrderFormat.ELECTRONIC_RESOURCE.toString())){
```

Comment:
Just curious why you cast format to String, but call `.toString()` on the OrderFormat...  both are acceptable, but why not be consistent?

Reply:
Changed it to be consistent.